src/commands/environments/create.ts

```diff
@@ -1,30 +1,98 @@
+import { Flags } from '@oclif/core'
+import inquirer from 'inquirer'
 import {
     createEnvironment,
-    CreateEnvironmentParams
+    environmentTypes
 } from '../../api/environments'
 import {
     descriptionPrompt,
     keyPrompt,
     namePrompt,
     environmentTypePrompt
 } from '../../ui/prompts'
-import CreateCommand from '../createCommand'
+import Base from '../base'
 
-export default class CreateEnvironment extends CreateCommand<CreateEnvironmentParams> {
+export default class CreateEnvironment extends Base {
     static hidden = false
     static description = 'Create a new Environment for an existing Feature.'
-    
-    prompts = [
-        keyPrompt,
-        namePrompt,
-        descriptionPrompt,
-        environmentTypePrompt,
-    ]
+
+    static flags = {
+        ...Base.flags,
+        'key': Flags.string({
+            description: 'Unique ID to refer to the environment.'
+        }),
+        'name': Flags.string({
+            description: 'Human readable name of the environment.',
+        }),
+        'description': Flags.string({
+            description: 'Description for display in the dashboard.',
+        }),
+        'type': Flags.string({
+            description: 'The type of environment',
+            options: environmentTypes
+        }),
+    }
 
     public async run(): Promise<void> {
+        const { flags } = await this.parse(CreateEnvironment)
```

Comment:
could we not move this to the CreateCommand and skip the prompt if the flag value is set or if the headless flag is set?

Reply:
the command should handle returning any errors if the required fields are missing, but the common create prompt should be able to handle parsing and returning the parameter values